v2/yarpcthrift/inbound_test.go

```diff
@@ -23,311 +23,65 @@ package yarpcthrift
-	}
-	h := unaryTransportHandler{Protocol: proto, ThriftHandler: handler}
-	res, resBody, err := unaryCall(ctx, h)
-
-	assert.Nil(t, res)
-	assert.Nil(t, resBody)
-	require.Error(t, err, "expected an error")
-	assert.Contains(t, err.Error(), "encode response error")
-
-}
-
-func TestDecodeEnvelopedError(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	proto := thrifttest.NewMockProtocol(mockCtrl)
-	// XXX DecodeEnveloped returns error
-	proto.EXPECT().DecodeEnveloped(gomock.Any()).Return(wire.Envelope{}, fmt.Errorf("decode enveloped error"))
-
-	ctx, cancel := context.WithTimeout(context.Background(), internaltesttime.Second)
-	defer cancel()
-
-	h := unaryTransportHandler{Protocol: proto, Enveloping: true}
-	res, resBody, err := unaryCall(ctx, h)
-
-	assert.Nil(t, res)
-	assert.Nil(t, resBody)
-	require.Error(t, err, "expected an error")
-	assert.Contains(t, err.Error(), "decode enveloped error")
-
-}
-
-func TestDecodeEnvelopedEnvelopeTypeError(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	proto := thrifttest.NewMockProtocol(mockCtrl)
-	// XXX DecodeEnveloped returns OneWay instead of expected Call
-	proto.EXPECT().DecodeEnveloped(gomock.Any()).Return(wire.Envelope{Type: wire.OneWay}, nil)
-
-	ctx, cancel := context.WithTimeout(context.Background(), internaltesttime.Second)
-	defer cancel()
-
-	h := unaryTransportHandler{Protocol: proto, Enveloping: true}
-	res, resBody, err := unaryCall(ctx, h)
-
-	assert.Nil(t, res)
-	assert.Nil(t, resBody)
-	require.Error(t, err, "expected an error")
-	assert.Contains(t, err.Error(), "unexpected envelope type: OneWay")
-
-}
-
-func TestDecodeNotEnvelopedError(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	proto := thrifttest.NewMockProtocol(mockCtrl)
-	// XXX Mocked decode returns decode error
-	proto.EXPECT().Decode(gomock.Any(), wire.TStruct).Return(wire.Value{}, fmt.Errorf("decode error"))
-
-	ctx, cancel := context.WithTimeout(context.Background(), internaltesttime.Second)
-	defer cancel()
-
-	h := unaryTransportHandler{Protocol: proto}
-	res, resBody, err := unaryCall(ctx, h)
-
-	assert.Nil(t, res)
-	assert.Nil(t, resBody)
-	require.Error(t, err, "expected an error")
-	assert.Contains(t, err.Error(), "decode error")
-
-}
-
-func TestUnaryHandlerError(t *testing.T) {
-	mockCtrl := gomock.NewController(t)
-	defer mockCtrl.Finish()
-
-	proto := thrifttest.NewMockProtocol(mockCtrl)
-	proto.EXPECT().Decode(gomock.Any(), wire.TStruct).Return(wire.Value{}, nil)
-
-	ctx, cancel := context.WithTimeout(context.Background(), internaltesttime.Second)
-	defer cancel()
-
-	handler := func(ctx context.Context, w wire.Value) (Response, error) {
-		// XXX returns error
-		return Response{}, fmt.Errorf("unary handler error")
+func TestEncodingHandler(t *testing.T) {
+	tests := []struct {
+		reqBody       interface{}
+		retResponse   Response
+		retError      error
+		expectedError string
```

Comment:
optional: We could also use `give` and `want` prefixes to distinguish what is provided from what is expected.